javagen/src/test/java/com/azure/autorest/MockUnitJavagen.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.azure.autorest;
+
+import com.azure.autorest.extension.base.jsonrpc.Connection;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+
+public class MockUnitJavagen extends Javagen{
+
+    public MockUnitJavagen() {
+        super(new Connection(System.out, System.in), "dummy", "dummy");
+        JavaSettingsAccessor.setHost(this);
+    }
+
+    @Override
+    public String readFile(String fileName) {
+        try {
+            InputStream fis = this.getClass().getClassLoader().getResourceAsStream(fileName);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(fis));
+            StringBuilder sb = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                sb.append(line);
+                sb.append("\n");
+            }
+            return sb.toString();
```

Comment:
Curious on what will be the difference between this and below
```
        try (InputStream inputStream = ResourceParserTests.class.getClassLoader().getResourceAsStream(filename);
             Reader in = new InputStreamReader(inputStream, StandardCharsets.UTF_8)) {
            int charsRead;
            while ((charsRead = in.read(buffer, 0, buffer.length)) > 0) {
                out.append(buffer, 0, charsRead);
            }
            return out.toString();
```

Reply:
Oh, I forgot to close the resource...

Reply:
Actually I'm surprised `\n` actually works in windows environment instead of `\r\n`.

Reply:
`\n` is probably fine, as snakeyaml likely able to handle both.

SDK usually does not know about current system, so it likely had to handle both.